src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,266 @@
+      }
+    } else if (attrName === 'disabled' && newValue !== oldValue) {
+      if (newValue == null) {
+        this.enable();
+      } else {
+        this.disable();
+      }
+    }
+  }
+
+  connectedCallback() {
+    if (!this.hasAttribute('disabled')) {
+      this.enable();
+    }
+
+    this.setAttribute('role', 'listbox');
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.associateElement?.(this);
+    }
+  }
+
+  disconnectedCallback() {
+    this.disable();
+
+    const mediaControllerId = this.getAttribute(
+      MediaUIAttributes.MEDIA_CONTROLLER
+    );
+    if (mediaControllerId) {
+      const mediaControllerEl = document.getElementById(mediaControllerId);
+      mediaControllerEl?.unassociateElement?.(this);
+    }
+  }
+
+  get keysUsed() {
+    return ['Enter', ' ', 'ArrowDown', 'ArrowUp', 'Home', 'End'];
+  }
+
+  #getItem(e) {
+    if (e.target === this) return;
+
+    const composedPath = e.composedPath();
+    const slotIndex = composedPath.findIndex(el => el.nodeName === 'SLOT');
+
+    return composedPath[slotIndex - 1];
+  }
+
+  handleSelection(e) {
+    const item = this.#getItem(e);
+
+    if (!item) return;
+
+    const selected = item.getAttribute('aria-selected') === 'true';
+
+    const slot = this.shadowRoot.querySelector('slot');
+
+    if (this.getAttribute('aria-multiselectable') !== 'true') {
+      slot.assignedElements().forEach(el => el.removeAttribute('aria-selected'));
+    }
+
+    if (selected) {
+      item.removeAttribute('aria-selected');
+    } else {
+      item.setAttribute('aria-selected', 'true');
+    }
+  }
+
+  handleMovement(e) {
+    const { key } = e;
+
+    let currentOption = this.#getItem(e);
+    if (!currentOption) {
+      const slot = this.shadowRoot.querySelector('slot');
+      currentOption = slot.assignedElements()
+        .filter(el => el.getAttribute('tabindex') === '0')[0];
+    }
+
+    let nextOption;
+    
+    switch (key) {
+      case 'ArrowDown':
+        nextOption = currentOption.nextElementSibling;
+        break;
+      case 'ArrowUp':
+        nextOption = currentOption.previousElementSibling;
+        break;
+      case 'Home':
+        nextOption = this.shadowRoot.querySelector('slot').assignedElements().shift();
+        break;
+      case 'End':
+        nextOption = this.shadowRoot.querySelector('slot').assignedElements().pop();
+        break;
+      default:
+        nextOption = this.#searchItem(key);
+        break;
+    }
+
+    if (nextOption) {
+      const slot = this.shadowRoot.querySelector('slot');
+      slot.assignedElements().forEach(el => el.setAttribute('tabindex', '-1'));
+      nextOption.setAttribute('tabindex', '0');
+      nextOption.focus();
+    }
+  }
+
+  handleClick(e) {
+    const item = this.#getItem(e);
+
```

Comment:
nit (nb): newline

Reply:
Whitespace warrior Wes. :)

I don't know why, but anything sitting right on top of an if() feels wrong to me. Gary and I have clearly worked in the same codebases. :)

Reply:
> Gary and I have clearly worked in the same codebases. :)

Yeah, we might've 😆